ui/src/components/Videos/index.js

```diff
@@ -1,48 +1,62 @@
-import React from "react";
-import { CSSTransition, TransitionGroup } from "react-transition-group";
-
+import React, { Component } from "react";
 import { StyledVideos } from "./style.css";
+import SearchResults from "react-filter-search";
 
-const Videos = ({ videos }) => {
-    const list = Object.keys(videos);
-    const preloadVal = list.length > 100 ? "none" : "auto";
-    return (
-        <StyledVideos>
-            <div className="section-title">Videos</div>
-            <TransitionGroup className={`videos__list videos__list_count-${list.length}`}>
-                {list.length &&
-                    list.map(video => {
-                        const src = "/video/" + videos[video];
-                        return (
-                            <CSSTransition key={video} timeout={500} classNames="videos-container_state" unmountOnExit>
-                                <div className="videos-container">
-                                    <div className="video-cap video-cap__name" title={videos[video]}>
-                                        {videos[video]}
-                                    </div>
-                                    <video controls preload={preloadVal}>
-                                        <source src={src} type="video/mp4" />
-                                    </video>
-                                </div>
-                            </CSSTransition>
-                        );
-                    })}
-            </TransitionGroup>
-            {
-                <CSSTransition
-                    in={!list.length}
-                    timeout={500}
-                    exit={false}
-                    classNames="videos__no-any_state"
-                    unmountOnExit
-                >
-                    <div className="no-any">
-                        <div title="No any" className="icon dripicons-hourglass" />
-                        <div className="nosession-any-text">NO VIDEOS YET :'(</div>
-                    </div>
-                </CSSTransition>
-            }
-        </StyledVideos>
-    );
-};
+export default class Videos extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: [],
+            value: "",
+        };
+    }
+
+    componentWillMount() {
+        const vids = [];
+        this.props.videos.forEach(function(item, index) {
+            vids.push({ name: item });
+        });
 
-export default Videos;
+        this.setState({ data: vids });
+    }
+
+    handleChange = event => {
+        const { value } = event.target;
+        this.setState({ value });
```

Comment:
you can use useState hook from React for the state management, that would make the component stateless on its own

Reply:
i followed the example from here:

Reply:
mb I can push to your branch? But you can spot some examples in the code on how the hook works

Reply:
Sure feel free to change anything you want